File: surf-friends/surf-friends-velocity/src/main/java/dev/slne/surf/friends/velocity/command/subcommand/friend/FriendToggleCommand.java
package dev.slne.surf.friends.velocity.command.subcommand.friend;

import dev.jorel.commandapi.CommandAPICommand;

import dev.slne.surf.friends.api.FriendApi;
import dev.slne.surf.friends.velocity.VelocityInstance;

public class FriendToggleCommand extends CommandAPICommand {
    public FriendToggleCommand(String name) {
        super(name);

        executesPlayer((player, info)-> {
            VelocityInstance.getInstance().getApi().toggle(player.getUniqueId());
        });
    }
}

File: surf-friends/src/main/java/dev/slne/surf/social/friend/command/friend/FriendToggleCommand.kt
package dev.slne.surf.social.friend.command.friend

import com.github.shynixn.mccoroutine.velocity.launch
import com.velocitypowered.api.proxy.Player
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.executors.CommandArguments
import dev.jorel.commandapi.executors.PlayerCommandExecutor
import dev.slne.surf.social.friend.SurfFriends
import dev.slne.surf.social.friend.command.argument.PlayerArgument
import dev.slne.surf.social.friend.player.FriendPlayer
import dev.slne.surf.social.friend.pluginContainer
import dev.slne.surf.social.friend.util.MessageBuilder

class FriendToggleCommand(commandName: String) : CommandAPICommand(commandName) {
    init {
        withPermission("surf.social.friend.command.send")
        executesPlayer(PlayerCommandExecutor { player: Player, _: CommandArguments ->
            pluginContainer.launch {
                val user = FriendPlayer.getPlayer(player.uniqueId)

                user.toggleNotifications()

                if(user.wouldLikeToBeNotified) {
                    SurfFriends.send(player, MessageBuilder().primary("Du hast die Freundschaftsbenachrichtigungen ").info("aktiviert").primary("."))
                } else {
                    SurfFriends.send(player, MessageBuilder().primary("Du hast die Freundschaftsbenachrichtigungen ").error("deaktiviert").primary("."))
                }
            }
        })
    }
}
